### PR TITLE
Increase the MSRV to 1.53.0

### LINKED ISSUE #514: Fuzz Testing & Nightly Cargo udeps are failing due to prost-derive
As mentioned, Fuzz Testing & Nightly Cargo udeps are failing due to prost-derive.

```
   Compiling prost-derive v0.6.1
error[E0034]: multiple applicable items in scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/prost-derive-0.6.1/src/lib.rs:111:14
    |
111 |             .intersperse(quote!(|));
    |              ^^^^^^^^^^^ multiple `intersperse` found
    |
    = note: candidate #1 is defined in an impl of the trait `Iterator` for the type `Map<I, F>`
    = note: candidate #2 is defined in an impl of the trait `Itertools` for the type `T`
help: disambiguate the associated function for candidate #1
    |
107 ~         let tags = Iterator::intersperse(field
108 +             .tags()
109 +             .into_iter()
110 +             .map(|tag| quote!(#tag)), {
111 +         let mut _s = $crate::__private::TokenStream::new();
112 +         $crate::quote_each_token!(_s $($tt)*);
  ...
help: disambiguate the associated function for candidate #2
    |
107 ~         let tags = Itertools::intersperse(field
108 +             .tags()
109 +             .into_iter()
110 +             .map(|tag| quote!(#tag)), {
111 +         let mut _s = $crate::__private::TokenStream::new();
112 +         $crate::quote_each_token!(_s $($tt)*);
```

### PR DESCRIPTION
**Pull request by hug-dev**
Also upgrades the prost and spiffe version to remove cargo audit
vulnerabilities. Removes the ignored ones.
Un-comment the CI checks that were commented before.

Fix #514

### CODE REVIEW

File: .cargo/audit.toml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 [advisories]
-ignore = ["RUSTSEC-2021-0073", # We do not use `prost_types::Timestamp` anywhere in our code.
-          "RUSTSEC-2020-0036"] # We don't have control over the exact dependencies of `protoc-grpcio`; See https://github.com/mtp401/protoc-grpcio/issues/36
+ignore = []
```
Comment (ionut-arm) [inline .cargo/audit.toml:2]: 🎉

File: ci.sh [hunk 2]
```diff
@@ -249,12 +249,12 @@ if [ "$PROVIDER_NAME" = "cargo-check" ]; then
...
-    # The oldest is currently in openSUSE Leap 15.3 and is 1.43.0.
+    # The oldest is currently in openSUSE Leap 15.3 and is 1.53.0.
     rustup update
 
-    rustup toolchain install 1.43.0
-    # The "jwt-svid-authenticator" can not be compiled on 1.43.0
-    RUST_BACKTRACE=1 cargo +1.43.0 check --release --features=all-providers,direct-authenticator,unix-peer-credentials-authenticator
+    rustup toolchain install 1.53.0
+    # The "jwt-svid-authenticator" can not be compiled on 1.53.0
+    RUST_BACKTRACE=1 cargo +1.53.0 check --release --features=all-providers,direct-authenticator,unix-peer-credentials-authenticator
```
Comment (ionut-arm) [inline ci.sh:257]: 🚀

**Review by ionut-arm — APPROVED** [review]
Awesome, so good to see this!

**Comment by ionut-arm** [comment]
Ahh, fuzz checks still failing, I think a `cargo update` is needed in the `fuzz` folder

**Comment by hug-dev** [comment]
I think it's because it's using the client before the PR we made today that fixes the issue there

**Review by anta5010 — APPROVED** [review]

**Pull request merged**